### ISSUE TITLE
scss included files are missing in webpack dependencies

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#15398]
When compiling Cockpit, changing a file, and building again, it does not work.

1. Build Cockpit
2. Change something in `pkg/lib/ct-card.scss` (for example)
3. Run `make` again
4. Nothing happens

I've been having to remove dist to build again... which is inefficient and slow. (Worse yet is when I forget to remove dist, reload the page, and wonder why my changes don't show up.)

I've had this issue with both the Machines and Storage pages, not necessarily with that file (but with others, such as `pkg/lib/patternfly/patternfly-4-overrides.scss`).

This is also a problem when running `tools/webpack-watch storaged`.

I'm pretty sure this _used to_ work, but I've had this for at least the past few weeks.

**Comment by martinpitt** [comment]
I started from a clean git checkout, then provided bots and built:

    ln -s ../bots
    ./autogen.sh --prefix=/usr  --enable-debug
    make -j4

Now running `make` again correctly does not do anything. Changing something in a package dir does rebuild:
```
$ touch pkg/storaged/details.jsx
$ make
make  all-am
make[1]: Entering directory '/var/home/martin/upstream/cockpit'
  WEBPACK  storaged
```
but indeed `touch pkg/lib/ct-card.scss` does not. That file is missing in dist/storaged/Makefile.deps, even though it is indirectly imported:

    pkg/storaged/storage.scss:@import "../lib/ct-card.scss";

Direct imports work, e.g. pkg/lib/journal.css or pkg/lib/table.css are in Makefile.deps, i.e. changes to it rebuild the webpack.


To work around whatever breaks on your machine, you can always build a single webpack only with either of these:

 - `make WEBPACK_PACKAGES=storaged`
 - `tools/webpack-watch storaged` and Control-C it after it's done, in case it does not pick up changes.

This avoids building *everything* all the time. This is also the prefered method for very generic changes like ct-card, as when this is done correctly, it would still rebuild most/all webpacks. This is usually not what you want as it's too annoying.

**Comment by martinpitt** [comment]
@garrett: Fixed in PR #15399. This fixes both `make` and `webpack-watch`. For not going insane I'd recommend to always use webpack-watch (once it's fixed, of course), as `make` will rebuild a lot after changing such central files.
